statsmodels/stats/anova.py

```diff
@@ -336,7 +361,7 @@ def anova_lm(*args, **kwargs):
     n_models = len(args)
     pr_test = "Pr(>%s)" % test
     names = ['df_resid', 'ssr', 'df_diff', 'ss_diff', test, pr_test]
-    table = DataFrame(np.zeros((n_models, 6)), columns = names)
+    table = DataFrame(np.zeros((n_models, 6)), columns=names)
```

Comment:
Good change.  This is about the only chance we get to make flake8 fixups.  There are a couple below (366, 376) that would be nice to get if convenient.  Also the `lmap`s on 369-370 can be made much more pythonic with a list comprehension.

Reply:
The try/except on 353-356 is heinous, should be made sane

Reply:
I will get those fixups.

I did not want to stray too far from my initial scope goal, but it seems instead of returning an error, a warning could be issued, the type changed to type I anova and solved to handle the user input error a little more gracefully.

Reply:
I think we should not automatically switch anova type, or at least it won't be a PR that will be quickly merged.
The warning is not enough indication that we switch methods. (I don't like it in general that when the user asks for A, then we give him/her B).

If there are ever other types in this case, then it will be messy and breaking backwards compatibility to change this.

However, if this is the only typ that makes sense, then we would need to specify that typ is ignored for multi-model comparisons. In that case raising just a warning would be enough

Reply:
Commit https://github.com/statsmodels/statsmodels/pull/5233/commits/8bccf95cc95e5168c7ade7ec91faadea8a148e27 should address these.